src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/product_v2/product.yml

```diff
@@ -65,3 +65,12 @@ admin_products_v2_delete:
             id_product: productId
     requirements:
         productId: \d+
+
+admin_products_v2_search_associations:
+    path: /search/{languageCode}
+    methods: [GET, POST]
+    defaults:
+        _controller: PrestaShopBundle\Controller\Admin\Sell\Catalog\Product\ProductController::searchAssociationsAction
+        _legacy_controller: AdminProducts
+    requirements:
+        languageCode: !php/const PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\TagIETF::IETF_TAG_REGEXP
```

Comment:
Why not use the value in the controller if this doesn't change ?

Reply:
There is no such const in the controller, besides the idea here is to have a single source of truth for the expected format of language code At first I had hard-coded it, but as @matks mentioned if we change this rule we would need to update it in multiple places

So having a single const is a better approach, and storing it inside the VO seems like the best candidate for this

Reply:
I mean, why not use TagIETF::IETF_TAG_REGEXP directly in the controller

Reply:
Ooh, because the requirements in routing are defined inside the yml, not in the controller
This enforces to respect the expected format or else the Router matcher doesn't recognize the route And when you try to generate with an invalid format an exception is thrown as well